### PR TITLE
Feature/assignment16

### PR DESCRIPTION
**Pull request by dheerajsrinath**
Create an abstract class WillDoEverthing in that create one method sum(int a), Call that sum()
method from Class Yes1 and do addition of three numbers.
Note - Use abstraction concept

### CODE REVIEW

File: src/test/java/Assignment16Test.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+public class Assignment16Test {
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Test
+    public void testAbstractClass() {
+        int integerToSum = 1;
+        Yes1 yes1 = new Yes1();
+        logger.info(Integer.toString(yes1.sum(integerToSum)));
+        Assertions.assertTrue(yes1.sum(integerToSum) == (1 + 1 + 1),
```
Comment (shrikanttechincal) [inline src/test/java/Assignment16Test.java]: you can write directly 3

File: src/test/java/Assignment16Test.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import org.slf4j.LoggerFactory;
+import com.assignment.*;
+
+public class Assignment16Test {
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Test
+    public void testAbstractClass() {
+        int integerToSum = 1;
+        Yes1 yes1 = new Yes1();
```
Comment (shrikanttechincal) [inline src/test/java/Assignment16Test.java]: remove this line,use new Yes1().sum(integerToSum) directly

File: src/test/java/Assignment16Test.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.assignment.*;
+
+public class Assignment16Test {
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Test
+    public void testAbstractClass() {
```
Comment (shrikanttechincal) [inline src/test/java/Assignment16Test.java]: use meaningful name ie ur test name should reflect what operation you are performing ie related to addition or sum

File: src/test/java/Assignment16Test.java (hunk 2 above)
Comment (dheerajsrinath) [reply src/test/java/Assignment16Test.java]: i cannot do that unless i declare Yes1 as static

File: src/test/java/Assignment16Test.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import com.assignment.*;
+
+public class Assignment16Test {
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Test
+    public void testAdditionUsingAbstract() {
+        int integerToSum = 1;
+        AdditionOfNumbers additionOfNumbers = new AdditionOfNumbers();
+        logger.info(Integer.toString(additionOfNumbers.getSum(integerToSum)));
```
Comment (PranavHanwante) [inline src/test/java/Assignment16Test.java:15]: intent+ ctrl+L

File: src/main/java/com/assignment/AdditionOfNumbers.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.assignment;
+
+abstract class WillDoEverything {
+    public abstract int getSum(int a);
+}
+
+public class AdditionOfNumbers extends WillDoEverything {
+    public int getSum(int a) {
```
Comment (PranavHanwante) [inline src/main/java/com/assignment/AdditionOfNumbers.java]: intent

File: src/test/java/Assignment16Test.java (hunk 4 above)
Comment (PranavHanwante) [inline src/test/java/Assignment16Test.java]: intent

**Review by PranavHanwante — APPROVED** [review]

**Review by akhil7000 — APPROVED** [review]

**Pull request merged**